src/vm/contexts.rs

```diff
@@ -437,17 +441,19 @@ impl <'a> GlobalContext<'a> {
         self.asset_maps.len() == 0
     }
 
-    pub fn log_asset_transfer(&mut self, sender: &PrincipalData, contract_name: &ContractName, asset_name: &ClarityName, transfered: Value) {
-        let asset_identifier = AssetIdentifier { contract_name: contract_name.clone(),
-                                                 asset_name: asset_name.clone() };
+    pub fn log_asset_transfer(&mut self, sender: &PrincipalData, contract_identifier: &QualifiedContractIdentifier, asset_name: &str, transfered: Value) {
+        let asset_name = String::from(asset_name).try_into().expect("Failed to get asset name");
+        let asset_identifier = AssetIdentifier { contract_identifier: contract_identifier.clone(),
+                                                 asset_name };
         self.asset_maps.last_mut()
             .expect("Failed to obtain asset map")
             .add_asset_transfer(sender, asset_identifier, transfered)
     }
 
-    pub fn log_token_transfer(&mut self, sender: &PrincipalData, contract_name: &ContractName, asset_name: &ClarityName, transfered: i128) -> Result<()> {
-        let asset_identifier = AssetIdentifier { contract_name: contract_name.clone(),
-                                                 asset_name: asset_name.clone() };
+    pub fn log_token_transfer(&mut self, sender: &PrincipalData, contract_identifier: &QualifiedContractIdentifier, asset_name: &str, transfered: i128) -> Result<()> {
```

Comment:
The `asset_name` parameter should just be &ClarityName rather than `&str`. Then you won't need to do the conversions and checks in 454.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1105/commits/a76e6119e4cf3284bc04d63cfebd20b156c1ced0

Reply:
Can you change `log_asset_transfer` as well ?

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1105/commits/fd249be6683ab9f81cea89c31c9a96a1290dc430